src/index.js

```diff
@@ -470,10 +478,11 @@ export default class HyperScreen extends React.Component {
    * preload screen and URL to load.
    */
   componentWillReceiveProps(nextProps) {
-    const newUrl = nextProps.navigation.state.params.url;
-    const oldUrl = this.props.navigation.state.params.url;
-    const newPreloadScreen = nextProps.navigation.state.params.preloadScreen;
-    const oldPreloadScreen = this.props.navigation.state.params.preloadScreen;
+    const { params } = this.getNavigationState();
+    const newUrl = params.url;
+    const oldUrl = params.url;
```

Comment:
Looks like `oldUrl` and `newUrl` point to same thing. Looks like the earlier code picked them differently from the current and next props.

Reply:
good catch

Reply:
addressed in f5396f3